addons/mail/models/mail_activity.py

```diff
@@ -577,6 +577,17 @@ def action_open_document(self):
             'view_mode': 'form',
         }
 
+    def action_snooze(self):
+        self.ensure_one()
```

Comment:
Is there a reason to keep the `ensure_one` constraint and force the recordset to hold a single record? Wouldn't it be better to remove that constraint and update the deadline of all records using a for loop?

That way, people do not have to write for loops from the outside. It's more convenient and work in any cases :smiley:

Something like this?

```python
def action_snooze(self):
    today = date.today()
    for activity in self:
        activity.write({'date_deadline': timedelta(days=7) + (
                        today if activity.date_deadline < today \
                              else activity.date_deadline)})
```